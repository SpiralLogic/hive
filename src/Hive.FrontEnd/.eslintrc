{
  "plugins": [
    "@typescript-eslint",
    "react-hooks",
    "testing-library",
    "jest-dom",
    "unicorn"
  ],
  "env": {
    "browser": true,
    "es6": true
  },
  "extends": [
    "kentcdodds",
    "plugin:jest/recommended",
    "plugin:jest-dom/recommended",
    "plugin:prettier/recommended",
    "plugin:testing-library/react",
    "plugin:testing-library/dom",
    "prettier",
    "plugin:unicorn/recommended",
    "preact"
  ],
  "rules": {
    "unicorn/no-null": "off",
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": [
      "error"
    ]
  },
  "globals": {
    "Atomics": "readonly",
    "SharedArrayBuffer": "readonly",
    "h": true
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json",
    "jsx": true
  },
  "overrides": [
    {
      "files": [
        "**/*.jsx",
        "**/*.tsx"
      ],
      "rules": {
        "unicorn/filename-case": "off"
      }
    }
  ],
  "settings": {
    "react": {
      "pragma": "h"
    },
    "propWrapperFunctions": [
      "forbidExtraProps",
      {
        "property": "freeze",
        "object": "Object"
      }
    ],
    "linkComponents": [
      "Hyperlink",
      {
        "name": "Link",
        "linkAttribute": "to"
      }
    ]
  },
  "ignorePatterns": [
    "/node_modules/*/**",
    "/*.config.js",
    "/src/public/*/**",
    "**/__coverage__/**",
    "/src/index.tsx",
    "/src/index.html",
    "/build.js",
    "/inject.shim.js"
  ]
}
