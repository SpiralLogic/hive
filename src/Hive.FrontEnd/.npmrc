node-version=20.11.1