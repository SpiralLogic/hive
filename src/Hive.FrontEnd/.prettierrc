{
  "singleQuote": true,
  "printWidth": 110,
  "proseWrap": "always",
  "bracketSameLine": true,
  "endOfLine": "auto",
  "trailingComma": "es5",
  "tabWidth": 2
}
