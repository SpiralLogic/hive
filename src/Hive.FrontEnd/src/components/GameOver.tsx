import '../css/share.css';

import { FunctionComponent } from 'preact';

const GameOver: FunctionComponent<{ outcome: string }> = ({ outcome }) => (
  <>
    <p>{outcome}</p>
    <button autofocus title="New Game" onClick={() => window.location.assign(`/`)}>
      New Game
    </button>
  </>
);
GameOver.displayName = 'GameOver';
export default GameOver;
