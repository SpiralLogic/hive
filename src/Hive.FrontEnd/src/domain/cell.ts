import { HexCoordinates } from './hex-coordinates';
import { Tile } from './tile';

export type Cell = {
  readonly coords: HexCoordinates;
  tiles: Array<Tile>;
};
