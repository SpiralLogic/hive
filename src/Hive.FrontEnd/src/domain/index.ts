export * from './cell';
export * from './game-state';
export * from './hex-coordinate';
export * from './move';
export * from './player';
export * from './row';
export * from './tile';
export * from './creature';
