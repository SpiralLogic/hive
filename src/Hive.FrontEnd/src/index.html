<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="description" content="Hive online board game">
  <title>Hive</title>
  [css]
  [scripts]    
</head>
</html>
