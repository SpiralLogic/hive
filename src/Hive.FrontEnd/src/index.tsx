import { h, render } from 'preact';
import App from './components/App';
import { serverConnectionFactory } from './services';
import creatures from './svg/creatures.svg?raw';

import GameEngine from './services/game-engine';

document.body.insertAdjacentHTML('beforeend', `${creatures}`);
document.title = import.meta.env.HIVE_APP_TITLE;

const [, , gameId, currentPlayer] = window.location.pathname.split('/');
const engine = new GameEngine({ gameId, currentPlayer });
const properties = {
  engine,
  connectionFactory: serverConnectionFactory,
};

render(h(App, properties), document.body);

export { removeOtherPlayerMoves } from './utilities/hextille';
export { cellKey } from './utilities/hextille';
console.log(import.meta.env);
