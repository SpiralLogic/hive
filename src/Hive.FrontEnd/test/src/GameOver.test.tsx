import { render, screen } from '@testing-library/preact';
import userEvent from '@testing-library/user-event';
import GameOver from '../../src/components/GameOver';
import { GameStatus } from '../../src/domain';
import { mockLocation } from '../helpers';

const gameOutcomes: Array<GameStatus> = [
  'NewGame',
  'MoveSuccess',
  'AiWin',
  'Player0Win',
  'Player1Win',
  'MoveSuccessNextPlayerSkipped',
  'MoveInvalid',
  'GameOver',
  'Draw',
];

describe('<GameOver>', () => {
  it('redirects to base url when close button is pressed', async () => {
    const location = { assign: vi.fn() };
    const restore = mockLocation(location);
    render(<GameOver outcome="Draw" />);

    await userEvent.click(screen.getByRole('button'));
    expect(location.assign).toBeCalledWith('/');
    restore();
  });
});

describe('<GameOver> snapshots', () => {
  it.each(gameOutcomes)(`%s player 0 matches`, (gameOutcome) => {
    expect(render(<GameOver outcome={gameOutcome} />)).toMatchSnapshot();
  });

  it.each(gameOutcomes)(`%s player 1 matches`, (gameOutcome) => {
    expect(render(<GameOver outcome={gameOutcome} />)).toMatchSnapshot();
  });
});
