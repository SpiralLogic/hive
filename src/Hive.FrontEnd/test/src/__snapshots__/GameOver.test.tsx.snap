// Vitest Snapshot v1

exports[`<GameOver> snapshots > AiWin player 0 matches 1`] = `
<p>
  AiWin
</p>
`;

exports[`<GameOver> snapshots > AiWin player 1 matches 1`] = `
<p>
  AiWin
</p>
`;

exports[`<GameOver> snapshots > Draw player 0 matches 1`] = `
<p>
  Draw
</p>
`;

exports[`<GameOver> snapshots > Draw player 1 matches 1`] = `
<p>
  Draw
</p>
`;

exports[`<GameOver> snapshots > GameOver player 0 matches 1`] = `
<p>
  GameOver
</p>
`;

exports[`<GameOver> snapshots > GameOver player 1 matches 1`] = `
<p>
  GameOver
</p>
`;

exports[`<GameOver> snapshots > MoveInvalid player 0 matches 1`] = `
<p>
  MoveInvalid
</p>
`;

exports[`<GameOver> snapshots > MoveInvalid player 1 matches 1`] = `
<p>
  MoveInvalid
</p>
`;

exports[`<GameOver> snapshots > MoveSuccess player 0 matches 1`] = `
<p>
  MoveSuccess
</p>
`;

exports[`<GameOver> snapshots > MoveSuccess player 1 matches 1`] = `
<p>
  MoveSuccess
</p>
`;

exports[`<GameOver> snapshots > MoveSuccessNextPlayerSkipped player 0 matches 1`] = `
<p>
  MoveSuccessNextPlayerSkipped
</p>
`;

exports[`<GameOver> snapshots > MoveSuccessNextPlayerSkipped player 1 matches 1`] = `
<p>
  MoveSuccessNextPlayerSkipped
</p>
`;

exports[`<GameOver> snapshots > NewGame player 0 matches 1`] = `
<p>
  NewGame
</p>
`;

exports[`<GameOver> snapshots > NewGame player 1 matches 1`] = `
<p>
  NewGame
</p>
`;

exports[`<GameOver> snapshots > Player0Win player 0 matches 1`] = `
<p>
  Player0Win
</p>
`;

exports[`<GameOver> snapshots > Player0Win player 1 matches 1`] = `
<p>
  Player0Win
</p>
`;

exports[`<GameOver> snapshots > Player1Win player 0 matches 1`] = `
<p>
  Player1Win
</p>
`;

exports[`<GameOver> snapshots > Player1Win player 1 matches 1`] = `
<p>
  Player1Win
</p>
`;
